application/shared-webapp/ui/components/Toast.tsx

```diff
@@ -94,6 +119,56 @@ function Toast<T extends ToastContents>({ state, ...props }: Readonly<ToastProps
   const { toastProps, titleProps, closeButtonProps, descriptionProps } = useToast(props, state, ref);
   const { content } = props.toast;
 
+  // Track if the toast is being hovered or focused
+  const [isPaused, setIsPaused] = useState(false);
+  const timerRef = useRef<NodeJS.Timeout | null>(null);
+  const remainingTimeRef = useRef<number | null>(null);
+  const startTimeRef = useRef<number | null>(null);
+
+  // Handle auto-dismiss functionality with pause on hover/focus
+  useEffect(() => {
+    // Only set up auto-dismiss if content is a ToastOptions object with a duration
+    if (
+      content &&
+      typeof content === "object" &&
+      !isValidElement(content) &&
+      "duration" in content &&
+      typeof content.duration === "number"
+    ) {
+      // If we're not paused, start/resume the timer
+      if (!isPaused) {
+        // If we have remaining time from a pause, use that, otherwise use the full duration
+        const duration = remainingTimeRef.current !== null ? remainingTimeRef.current : content.duration;
+        startTimeRef.current = Date.now();
+
+        timerRef.current = setTimeout(() => {
+          toastQueue.close(props.toast.key);
+        }, duration);
+      } else if (timerRef.current && startTimeRef.current !== null) {
+        // If we're paused and have an active timer, clear it and calculate remaining time
+        clearTimeout(timerRef.current);
+        timerRef.current = null;
+
+        // Calculate how much time is left
+        const elapsedTime = Date.now() - startTimeRef.current;
+        remainingTimeRef.current = Math.max(0, content.duration - elapsedTime);
+      }
```

Comment:
I don't remember this logic - did you create the Toast component or use the https://github.com/tinytek-io/windcraft-starter-kit/blob/main/ui/components/Toast.tsx ?